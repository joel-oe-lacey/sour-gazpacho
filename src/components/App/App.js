import React, { Component } from 'react';
import './App.scss';
import Nav from '../Nav/Nav';
import LoadingPage from '../LoadingPage/LoadingPage';
import Homepage from '../Homepage/Homepage';
import LoginForm from '../LoginForm/LoginForm';
import MovieDetails from '../MovieDetails/MovieDetails'
import { connect } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { loadMovies } from '../../actions';
import { fetchData } from '../../utils/fetchCalls'

class App extends Component {
  componentDidMount() {
    fetchData('https://rancid-tomatillos.herokuapp.com/api/v1/movies')
      .then(response => response.json())
      .then(movieData => {
        this.props.loadMoviesToStore(movieData.movies)
      })
  }

  render() {
    return (
      <div className="app">
        <Route exact path="/login" component={LoginForm} />
        <Route exact path='/' render={() => {
          return (
            <section>
              <Nav />
              {!this.props.movies.length && <LoadingPage />}
              {this.props.movies.length && <Homepage allMovies={this.props.movies} />}
            </section>
          )}
        } />
        <Route exact path='/movies/:id' render={({ match }) => {
          const movieData = this.props.movies.find(movie => movie.id === parseInt(match.params.id))
          return (
            <section>
            <Nav /><MovieDetails movie={movieData} />
            </section>
          )}
        } />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  loadMoviesToStore: (movies) => { dispatch(loadMovies(movies)) }
});

const mapStateToProps = state => ({
  movies: state.movies,
})

export default connect(mapStateToProps, mapDispatchToProps)(App)
