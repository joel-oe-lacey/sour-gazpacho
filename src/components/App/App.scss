@import '../../variables';

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

#root,
body,
html,
.app {
  height: 100%;
  width: 100%;
  background-color: $primary-gray-color;
  color: $primary-white-color;
}
