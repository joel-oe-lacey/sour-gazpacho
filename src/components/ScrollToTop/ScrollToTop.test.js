import React from 'react';
import ScrollToTop from './ScrollToTop.js';
import App from '../App/App.js';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

describe('ScrollToTop', () => {
  let wrapper;
  let history;

  beforeEach(() => {
    wrapper = mount(
      <BrowserRouter initialEntries={['/']}>
        <ScrollToTop>
          <App />
        </ScrollToTop>
      </BrowserRouter>
    );
    history = wrapper.instance().history;
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should scroll to top if route pathname changes', () => {
    global.scrollTo = jest.fn();
    expect(global.scrollTo).not.toHaveBeenCalled();
    history.push('/new-url');
    expect(global.scrollTo).toHaveBeenCalledWith(0, 0);
  })

})
