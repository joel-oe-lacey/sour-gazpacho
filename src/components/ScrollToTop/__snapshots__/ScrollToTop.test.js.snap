// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop should match snapshot 1`] = `
<BrowserRouter
  initialEntries={
    Array [
      "/",
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(ScrollToTop)>
      <ScrollToTop
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      />
    </withRouter(ScrollToTop)>
  </Router>
</BrowserRouter>
`;
